server/src/main/java/org/elasticsearch/common/logging/HeaderWarning.java

```diff
@@ -302,11 +302,19 @@ private static char hex(int b) {
         }
     }
 
+    public static String getProductOrigin() {
+        return getSingleValue(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER);
+    }
+
     public static String getXOpaqueId() {
+        return getSingleValue(Task.X_OPAQUE_ID_HTTP_HEADER);
+    }
+
+    private static String getSingleValue(String productOrigin) {
```

Comment:
`productOrigin` seems like an odd name for this parameter given it appears that it should be taking a header name - maybe change to `headerName`?